internal/corazawaf/transaction.go

```diff
@@ -900,14 +900,19 @@ func (tx *Transaction) ProcessRequestBody() (*types.Interruption, error) {
 		return nil, nil
 	}
 
+	if tx.interruption != nil {
+		tx.WAF.Logger.Error("Calling ProcessRequestBody but there is a preexisting interruption")
+		return tx.interruption, nil
+	}
+
 	if tx.LastPhase >= types.PhaseRequestBody {
 		// Phase already evaluated
 		tx.WAF.Logger.Warn("ProcessRequestBody has already been called")
 		return tx.interruption, nil
 	}
 
-	if tx.interruption != nil {
-		tx.WAF.Logger.Error("Calling ProcessRequestBody but there is a preexisting interruption")
+	if tx.LastPhase != types.PhaseRequestHeaders {
```

Comment:
Isn't `tx.LastPhase != 1` same as  that `tx.LastPhase >= 2` (phases can be 1,2,3,4,5)? aren't both the same check in practice?

Reply:
It enforces that the header phase has been performed before the body phase.
For the request, it denies calling `ProcessRequestBody` as the first process executed ( `PhaseUnknown` (`0`) --> `PhaseRequestBody` (`2`)).
For the response, it denies calling `ProcessResponseBody` from any previous phase, but `PhaseResponseHeaders`.

What I'm facing is related to a "jump" from `PhaseUnknown` (`0`)to `PhaseResponseBody` (`4`), but I thought it worth adding symmetry in the code also for the process request checking for the expected order of the phases.

Reply:
I see, thanks for the explanation. In such case I would say, `if tx.LastPhase != 1` is enough check for this, no need for two separated checks.

Reply:
👍 Removed one `if` during normal circumstances but left some logic to not lose log details about what happened